### ISSUE TITLE
Add utf8 charset support

### ISSUE THREAD

**Issue by asluchevskiy** [flask-restful/flask-restful#552]
Hello, I just wrote a small piece of code to make all responses unicode.
Is it possible to make this default, or add as an option?

```python
from flask_restful import Api, output_json


class UnicodeApi(Api):
    def __init__(self, *args, **kwargs):
        super(UnicodeApi, self).__init__(*args, **kwargs)
        self.app.config['RESTFUL_JSON'] = {
            'ensure_ascii': False
        }
        self.representations = {
            'application/json; charset=utf-8': output_json,
        }
```

**Comment by asluchevskiy** [comment]
Also, it is nessessary to call `sys.setdefaultencoding` method before run the app, if your input json data will be unicode, otherwise you will get an `UnicodeEncodeError`.

```python
import sys
import settings
from app.api import app

if __name__ == '__main__':
    reload(sys)
    sys.setdefaultencoding('utf-8')
    app.run(debug=True)
```

**Comment by weaming** [comment]
+1 

Why not UTF-8? what's the difference between them

**Comment by jabbalaci** [comment]
Thanks a lot! UTF-8-encoded JSON output should be the _default_ IMO.

**Comment by justanr** [comment]
There's absolutely no reason to add a charset to `application/json`, from [RFC 7159](https://tools.ietf.org/html/rfc7159):

> 8.1.  Character Encoding

>   JSON text SHALL be encoded in UTF-8, UTF-16, or UTF-32.  The default
   encoding is UTF-8, and JSON texts that are encoded in UTF-8 are
   interoperable in the sense that they will be read successfully by the
   maximum number of implementations; there are many implementations
   that cannot successfully read texts in other encodings (such as
   UTF-16 and UTF-32).

>   Implementations MUST NOT add a byte order mark to the beginning of a
   JSON text.  In the interests of interoperability, implementations
   that parse JSON texts MAY ignore the presence of a byte order mark
   rather than treating it as an error.

Do note that SHALL is defined as "MUST This word, or the terms "REQUIRED" or "SHALL", mean that the definition is an absolute requirement of the specification." as per RFC 2119.

And from the Python json.dump/dumps documentation:

> encoding is the character encoding for str instances, default is UTF-8.

Adding a charset to application/json is counterproductive as that implied you could specify a different encoding (latin-1 for example) which will cause issues with JSON parsers that are standards complaint and don't take into account the unnecessary charset attribute (`requests` accepts it, though I'm not sure why)

**Comment by jabbalaci** [comment]
If I understand you correctly, JSON outputs are encoded in UTF-8, UTF-16, or UTF-32, but the default is UTF-8. With Flask-RESTful when I return something and open it in the browser, it's encoded as windows-1252. So it seems that windows-1252 is the default.

**Comment by synweap15** [comment]
Adding my two cents, it seems there's an issue in Chromium (affecting Chrome) with rendering UTF-8 chars without setting `charset=utf8` in the response headers which may cause the output to be rendered wrongly. I had a case where the output was displayed correctly in Firefox, but Chrome was causing problems. [Here](https://bugs.chromium.org/p/chromium/issues/detail?id=438464) is a related bug description. So you may have the data encoded correctly, but due to problems with the Chromium implementation itself, it's mistreated.

I understand that @justanr approach is absolutely correct in terms of sticking to RFC, yet @asluchevskiy 's approach allows to overcome mentioned problem if you don't want wait for Chromium to be patched ;)

**Comment by earlgreyness** [comment]
They seem to have fixed the issue on Jul 28, 2017. [Here](https://bugs.chromium.org/p/chromium/issues/detail?id=438464#c24) is the comment confirming the fix for Chromium version 62.0.3202.38. We should expect this patch to be included in Google Chrome as well (since they share the same codebase) in one of its next stable releases in the nearest future.

**Comment by phnessu4** [comment]
It's 2020, the safari 13.0.5 has the same issue, the ghost still here.
